docs/source/developer/tests.rst

```diff
@@ -39,67 +39,143 @@ If you need access to Windows you can download a
+    :emphasize-lines: 1
+
+    @terraform('file_example')
+    def test_file_example(file_example):
 
-The `Placebo <http://placebo.readthedocs.io/en/latest/>`_ library is used to
-record and replay AWS responses so that tests can run locally, and in a fraction
-of the time it would take to interact with live AWS services.
+        assert file_example['local_file.bar.content'] == 'bar!'
 
-In order to write a placebo test two helper methods are provided:
+Now when the test is run it will use the data previously recorded terraform resources and not run terraform directly.
+When committing your test, don't forget to include the ``tests/terraform/file_example`` directory!
 
-  - `record_flight_data` - use this when creating the test
-  - `replay_flight_data` - use this when the test is completed
+If your test performs destructive actions against a cloud resource created by terraform, check out `Controlling Resource Cleanup`_
 
-When first creating a test, use the `record_flight_data` method.  This will
-contact AWS and store all responses as files in the placebo directory
-(`tests/data/placebo/`).  The method takes one parameter, which is the directory
+Recording Custodian Interactions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Cloud Custodian tests provide a pytest fixture, ``test``, that provides access to
+common unitest methods (such as ``assertEqual``) as well as the placebo based test methods.
+In order to write a placebo enabled test two helper methods are provided:
+
+  - ``record_flight_data`` - use this when creating the test
+  - ``replay_flight_data`` - use this when the test is completed
+
+When first creating a test, use the ``record_flight_data`` method.  This will
+contact the cloud and store all responses as files in the placebo directory
+(``tests/data/placebo/``).  The method takes one parameter, which is the directory
 name to store placebo output in and it must be unique across all tests.  For
 example:
 
   .. code-block:: python
-    :emphasize-lines: 2,3
+    :emphasize-lines: 2
 
-    def test_example(self):
-        session_factory = self.record_flight_data(
-            'test_example')
+    def test_example(test):
+        session_factory = test.record_flight_data('test_example')
 
         policy = {
             'name': 'list-ec2-instances',
-            'resource': 'ec2'
+            'resource': 'aws.ec2',
         }
-            
-        policy = self.load_policy(
+
+        policy = test.load_policy(
             policy,
-            session_factory=session_factory)
+            session_factory=session_factory
+        )
 
         resources = policy.run()
-        self.assertEqual(len(resources), 1)
+        test.assertEqual(len(resources), 1)
 
 Now run this test using the following command to generate the placebo data:
 
   .. code-block:: shell
 
-    $ nosetests -s -v tests/path/to/test.py
+    $ pytest tests/path/to/test.py -s -v
 
 This may take a little while as the test is contacting AWS.
```

Comment:
Earlier you replace 'AWS' with 'the cloud', is it worth doing that here too?

Reply:
the example is pretty specific to AWS in this context so the explicit name seems fine.